agent/agentssh/agentssh.go

```diff
@@ -298,6 +305,35 @@ func extractMagicSessionType(env []string) (magicType MagicSessionType, rawType
 	})
 }
 
+// sessionCloseTracker is a wrapper around Session that tracks the exit code.
+type sessionCloseTracker struct {
+	ssh.Session
+	exitOnce sync.Once
+	code     atomic.Int64
+}
+
+var _ ssh.Session = &sessionCloseTracker{}
+
+func (s *sessionCloseTracker) track(code int) {
+	s.exitOnce.Do(func() {
+		s.code.Store(int64(code))
+	})
+}
+
+func (s *sessionCloseTracker) exitCode() int {
+	return int(s.code.Load())
+}
+
+func (s *sessionCloseTracker) Exit(code int) error {
+	s.track(code)
+	return s.Session.Exit(code)
+}
+
+func (s *sessionCloseTracker) Close() error {
+	s.track(1)
```

Comment:
What does `1` mean in this instance?

Reply:
Same as a shell script exiting with generic error (1). Calling `Close` instead of `Exit` on the session indicates an "error" state (not error as in log, but error as in session didn't go right).